From 81a8117b41586f07c43ad70cb5ca17532fdcfa7f Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Tue, 18 Sep 2012 20:52:35 +0200
Subject: MINOR: config: set the bind_conf entry on listeners created from a "listen" line.

Otherwise we would risk a segfault when checking the config's validity
(eg: when looking for conflicts on ID assignments).

Note that the same issue exists with peers_fe and the global stats_fe. All
listeners should be reviewed and simplified to use a compatible declaration
mode.
---
 src/cfgparse.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/src/cfgparse.c b/src/cfgparse.c
index 08c88d4..d2f0dcf 100644
--- a/src/cfgparse.c
+++ b/src/cfgparse.c
@@ -1447,12 +1447,17 @@ int cfg_parse_listen(const char *file, int linenum, char **args, int kwm)
 		/* parse the listener address if any */
 		if ((curproxy->cap & PR_CAP_FE) && *args[2]) {
 			struct listener *new, *last = curproxy->listen;
+
 			if (!str2listener(args[2], curproxy, file, linenum)) {
 				err_code |= ERR_FATAL;
 				goto out;
 			}
+
+			bind_conf = bind_conf_alloc(&curproxy->conf.bind, file, linenum, args[2]);
+
 			new = curproxy->listen;
 			while (new != last) {
+				new->bind_conf = bind_conf;
 				new = new->next;
 				global.maxsock++;
 			}
-- 
1.7.1

